match.c

```diff
@@ -361,9 +361,11 @@ ZLIB_INTERNAL unsigned longest_match(deflate_state *const s, IPos cur_match) {
     unsigned int wmask = s->w_mask;
 
     register unsigned char *strend = s->window + s->strstart + MAX_MATCH;
-    register uint16_t scan_start = *(uint16_t*)scan;
-    register uint16_t scan_end   = *(uint16_t*)(scan+best_len-1);
-
+    
```

Comment:
This empty line should be after "uint16_t scan_start, scan_end;" due to C requirement that all declarations should be before any of the variables are used. The empty line after variable declarations makes it clear where declarations stop and code starts.

Reply:
Makes sense, I just wanted to keep the relevant memcpy next to the variable, but I guess in strict C we can't organise code like that.

However, there are other issues in the code where variables and code is mixed. I will fix this one.